### PR TITLE
MAINT: Use PyArray_NewFromDescr where possible, remove unused definitions

### PR DESCRIPTION
**Pull request by eric-wieser**
A cleanup step in preparation for #11237

### CODE REVIEW

File: numpy/core/src/multiarray/calculation.c [hunk 1]
```diff
@@ -100,10 +100,10 @@ PyArray_ArgMax(PyArrayObject *op, int axis, PyArrayObject *out)
     }
 
     if (!out) {
-        rp = (PyArrayObject *)PyArray_New(Py_TYPE(ap), PyArray_NDIM(ap)-1,
-                                          PyArray_DIMS(ap), NPY_INTP,
-                                          NULL, NULL, 0, 0,
-                                          (PyObject *)ap);
+        rp = (PyArrayObject *)PyArray_NewFromDescr(
+                Py_TYPE(ap), PyArray_DescrFromType(NPY_INTP),
```
Comment (eric-wieser) [inline numpy/core/src/multiarray/calculation.c]: We're making the assumption that `PyArray_DescrFromType` does not return `NULL` here. This is already assumed elsewhere

File: numpy/core/src/multiarray/calculation.c [hunk 2]
```diff
@@ -100,10 +100,10 @@ PyArray_ArgMax(PyArrayObject *op, int axis, PyArrayObject *out)
     }
 
     if (!out) {
-        rp = (PyArrayObject *)PyArray_New(Py_TYPE(ap), PyArray_NDIM(ap)-1,
-                                          PyArray_DIMS(ap), NPY_INTP,
-                                          NULL, NULL, 0, 0,
```
Comment (eric-wieser) [inline numpy/core/src/multiarray/calculation.c]: Passing the dtype directly eliminates the need for `itemsize` to be passed (as 0).

File: numpy/core/src/multiarray/calculation.c (hunk 1 above)
Comment (charris) [reply numpy/core/src/multiarray/calculation.c]: I assume this is safe because it is a known numeric type?

Comment (eric-wieser) [reply numpy/core/src/multiarray/calculation.c]: Yep, `PyArray_DescrFromType` starts with an error-free lookup of builtin type numbers

**Comment by charris** [comment]
LGTM. What is the benefit of doing this?

**Comment by eric-wieser** [comment]
I want to add a `base` argument to the internal array constructor `PyArray_NewFromDescr_int`, to fix  #11237. Removing the indirection here makes that a little easier to do in my next pass

**Comment by charris** [comment]
Thanks Eric.

**Pull request merged**